### PR TITLE
ENH Gracefully handle AJAX failures

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-linkfield/issues/110

I tested by adding the following in to controller endpoints on LinkFieldController:
- `$this->jsonError(500, 'FFF');` - expected failure
- `throw new Exception('AAA');` - unexpected failure
- `set_time_limit(1);while(true);` - simulate timeout

I tested when:
- Reading links `LinkFieldController::getLinkData()`
- Deleting links `LinkFieldController::linkDelete()`
- Saving links `LinkFieldController::save()`

### CODE REVIEW

File: client/lang/en.js [hunk 1]
```diff
@@ -13,6 +13,13 @@ if (typeof(ss) === 'undefined' || typeof(ss.i18n) === 'undefined') {
     "LinkField.DELETE_ERROR": "Failed to delete link",
     "LinkField.ADD_LINK": "Add Link",
     "LinkField.ARCHIVE": "Archive",
-    "LinkField.DELETE": "Delete"
+    "LinkField.DELETE": "Delete",
+    "LinkField.LINK_DRAFT_TITLE": "Link has draft changes",
+    "LinkField.LINK_DRAFT_LABEL": "Draft",
+    "LinkField.LINK_MODIFIED_TITLE": "Link has unpublished changes",
```
Comment (emteknetnz) [inline client/lang/en.js:19]: There was a bunch of translations missing from the "dist" version of en.js, so copy pasted them in from the "src" version

Comment (ChloeSartorelli) [reply client/lang/en.js:19]: This isn't really necessary - when we finally do the standardisation of this module it'll be added to tx-translator's module set and we can do it all that way. This change is okay here but in future probably just leave it for when we're at that stage.

**Comment by GuySartorelli** [comment]
There are conflicts to resolve.

**Comment by emteknetnz** [comment]
Conflicts resolved

**Review by GuySartorelli — APPROVED** [review]
LGTM.

In a perfect world we'd add some JEST tests for this but I'm not too bothered about it, the main thing IMO is that the data doesn't end up in a bad state which it looks like it would be pretty hard to accomplish via thrown exceptions etc.

**Pull request merged**